python/nav/web/business/reportengine.py

```diff
@@ -0,0 +1,125 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2016 UNINETT AS
+#
+# This file is part of Network Administration Visualized (NAV).
+#
+# NAV is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License version 2 as published by
+# the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+# more details.  You should have received a copy of the GNU General Public
+# License along with NAV. If not, see <http://www.gnu.org/licenses/>.
+#
+"""Module for emailing status reports"""
+
+from __future__ import print_function
+
+import logging
+from collections import namedtuple
+from datetime import date, datetime, timedelta
+
+from django.core.mail import send_mail
+from django.template.loader import render_to_string
+
+from nav.web.business import utils
+from nav.models.profiles import ReportSubscription
+
+_logger = logging.getLogger(__name__)
+Report = namedtuple('Report', ['subject', 'period', 'text_message',
+                               'html_message'])
+
+
+def send_reports(period):
+    """Sends all reports for the given period
+
+    Supported periods are in the ReportSubscription class
+    """
+
+    report_types = [t for t, _ in ReportSubscription.TYPES]
+    for report_type in report_types:
+        _logger.debug('Sending reports for period %s, type %s',
+                      period, report_type)
+        report = build_report(period, report_type)
+        subscriptions = ReportSubscription.objects.filter(
+            period=period, report_type=report_type)
+        for subscription in subscriptions:
+            send_report(report, subscription.address.address)
+        _logger.info('%s %s availability: Sent %s reports',
+                     period, report_type, subscriptions.count())
+
+
+def send_report(report, to_address):
+    """Sends a single email report"""
+    _logger.debug('Sending report to %s', to_address)
+    send_mail(
+        report.subject,
+        report.text_message,
+        'noreply@example.com',
```

Comment:
Why hardcode the e-mail address, when there is a perfectly acceptable configuration option for sender address in `nav.conf`? :-)